presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcPageSource.java

```diff
@@ -74,23 +80,33 @@
     private int batchId;
     private long completedPositions;
     private boolean closed;
+    private boolean rowsDeleteLoad;
+    private Optional<BitSet> rowsDeleted = Optional.empty();
 
     public OrcPageSource(
+            OrcStorageManager orcStorageManager,
+            FileSystem fileSystem,
             Optional<ShardRewriter> shardRewriter,
             OrcBatchRecordReader recordReader,
             OrcDataSource orcDataSource,
             List<Long> columnIds,
             List<Type> columnTypes,
             List<Integer> columnIndexes,
             UUID shardUuid,
+            boolean tableSupportsDeltaDelete,
```

Comment:
@highker has some ideas on how to refactor out this flags with abstraction :)

Reply:
Let's chat in person how to abstract the page source out so that `getNextPage` logic in this class can remain almost untouched. The idea is to wrap around `recordReader` with a new `ConnectorPageSource` s.t., the new page source can decide if to perform a delta filter.